src/kafka/clients.clj

```diff
@@ -0,0 +1,94 @@
+(ns kafka.clients
+  (:gen-class)
+  (:import
+    (java.time Duration)
+    (java.net InetAddress)
+    (org.apache.kafka.common.serialization StringSerializer StringDeserializer)
+    (org.apache.kafka.clients.producer KafkaProducer ProducerRecord)
+    (org.apache.kafka.clients.consumer KafkaConsumer)))
+
+
+(def properties
+  (atom {:producer-config {"client.id",         (.getHostName (InetAddress/getLocalHost))
+                           "group.id",          "group1"
+                           "bootstrap.servers", "localhost:9092"
+                           "key.serializer"     StringSerializer
+                           "value.serializer"   StringSerializer}
+         :consumer-config {"client.id",         (.getHostName (InetAddress/getLocalHost))
+                           "group.id",          "group1"
+                           "bootstrap.servers", "localhost:9092"
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer}}))
+
+(def clients (atom {:producer [] :consumer []}))
+
+(defn build-producer
+  []
+  (let [producer (KafkaProducer. (get @properties :producer-config))
+        _        (swap! clients update :producer conj producer)]
+    producer))
+
+(defn build-consumer
+  []
+  (let [consumer (KafkaConsumer. (get @properties :consumer-config))
+        _        (.subscribe consumer ["example-topic"])
+        _        (swap! clients update :consumer conj consumer)]
+    consumer))
```

Comment:
These build fns should remain idempotent. Swapping with the atom should be separate fns